Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,61 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
+            var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
+            Console.WriteLine($"{args} : {result}");
+        }
+
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
+        public static int FindDigit(string equation)
+        {         
             // Add your code here.
-            throw new NotImplementedException();
+            //Splitting The equation into three variables eg. 42*47=1?74 to A=42 B=47 and C=1?74
+            string[] splitOnMultiply = equation.Split('*');
+            var A = splitOnMultiply[0];
+            string[] splitOnEqual = splitOnMultiply[1].Split('=');
+            var B = splitOnEqual[0];
+            var C = splitOnEqual[1];
+            var missingDIgitHolder = '?';
+            var result = -1;
+            var ans = 0.0;
+            string evaluatedEquation ="";
+
+            //Checking which string variable contain "?" mark and storing the evaluated result in ans
+            if(A.Equals("0") || B.Equals("0") || !equation.Contains("?"))
+                return result;
+            if(A.Contains(missingDIgitHolder))
+            {
+                ans = double.Parse(C)/double.Parse(B);
+                evaluatedEquation= ans.ToString() + "*" + B + "=" + C; 
+            }
+            else if(B.Contains(missingDIgitHolder))
+            {
+                ans = double.Parse(C)/double.Parse(A);
+                evaluatedEquation = A + "*" + ans.ToString() + "=" + C;
+            }
+            else if(C.Contains(missingDIgitHolder))
+            {
+                ans = double.Parse(A) * double.Parse(B);
+                evaluatedEquation = A + "*" + B + "=" + ans.ToString();
+            }
+            
+            //Comparing the original and evaluated variables and finding the missing value
+            if(equation.Length == evaluatedEquation.Length)

```

Comment:
Is there a better way of check for validity of solution rather than concatenating strings?

Reply:
Yes,  I implemented a function  which passes the value that contains '?' and the evaluated result.
In return it gives the missing digit